include/xrpl/protocol/STAmount.h

```diff
@@ -135,16 +141,46 @@ class STAmount final : public STBase, public CountedObject<STAmount>
     STAmount(A const& asset, int mantissa, int exponent = 0);
 
     template <AssetType A>
-    STAmount(A const& asset, Number const& number)
+    STAmount(
+        A const& asset,
+        Number const& number,
+        std::optional<Number::EnforceInteger> enforce = std::nullopt)
         : STAmount(asset, number.mantissa(), number.exponent())
     {
+        enforceConversion_ = enforce;
+        if (!enforce)
+        {
+            // Use the default conversion behavior
```

Comment:
I don't understand why we need this block. The `operator Number` does not seem to be doing any useful work when `enforceConversion_` is empty (or `none`)

Reply:
> I don't understand why we need this block. The `operator Number` does not seem to be doing any useful work when `enforceConversion_` is empty (or `none`)

If the default conversion uses an option that can throw, and the converted `Number` is invalid, that conversion will throw. Otherwise the result is unneeded. There may be a better way to do it.